packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -258,27 +229,9 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
   }
 
   async getConnectionInfo(): Promise<ConnectionInfo> {
-    // buildInfo try/catch can be removed after MONGOCRYPT-308
-    let buildInfo;
-    try {
-      buildInfo = await this.runCommandWithCheck('admin', {
-        buildInfo: 1
-      }, this.baseCmdOptions);
-    } catch (e: unknown) {
-      if ((e as Error)?.message.includes('not supported for auto encryption')) {
```

Comment:
Should we have been re-throwing this error when it didn't match before?

Reply:
I think so, yes, otherwise this would have ended up breaking the calls inside `getConnectInfo()`. I think this is a good way to resolve this tho :)

Reply:
Ah wait, this is failing CI because now `buildInfo` does fail in API-strict mode. I’ll take a look in more detail, but we may just want to add it to the other calls in the `Promise.all()` below and handle it the same way